app/src/main/java/com/example/bill/groceryapp/MainActivity.java

```diff
@@ -0,0 +1,103 @@
+package com.example.bill.groceryapp;
+
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ListView;
+
+import java.util.ArrayList;
+
+public class MainActivity extends AppCompatActivity {
+
+    private Button addButton;
+    private EditText newItemInput;
+    private ArrayList<String> items = new ArrayList<>();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        // Setup List
+        final ArrayAdapter<String> itemsAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, items);
+        final ListView groceryList = (ListView) findViewById(R.id.list);
+        groceryList.setAdapter(itemsAdapter);
+        // Add Button
+        addButton = (Button) findViewById(R.id.button);
+        // Where user can type to add new items
+        newItemInput = (EditText) findViewById(R.id.editText);
+
+        // Add the new item when the addButton is clicked and clear the input field
+        addButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                items.add(newItemInput.getText().toString());
+                newItemInput.setText("");
+                itemsAdapter.notifyDataSetChanged();
+            }
+        });
+
+        // Show an AlertDialog when an item is clicked
+        groceryList.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, final int position, long id) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
+                builder.setTitle("Edit Item");
+
+                // Input text field to edit item
+                final EditText input = new EditText(MainActivity.this);
+                input.setText(parent.getItemAtPosition(position).toString());
```

Comment:
To do this without getting parent view (which might break if you change the UI), can't you use MainActivity.this.items.get(position)?